### PR TITLE
fix: crash on Async dialog not being attached to a context

### LINKED ISSUE #13795: 2.16 Bug: crash on Async dialog not being attached to a context
Similar to #13085 

Cause: https://github.com/ankidroid/Anki-Android/commit/6ea36860a8e9fddadff0bf484746248fd843eb53

```
java.lang.IllegalStateException: Fragment SyncErrorDialog{31dcea2} (bfb1ab5d-fa6d-4d4b-9fe6-9bf9ea0489c8 tag=dialog) not attached to a context.
	at androidx.fragment.app.Fragment.requireContext(Fragment.java:32)
	at androidx.fragment.app.Fragment.getResources(Fragment.java:1)
	at com.ichi2.anki.dialogs.SyncErrorDialog.getTitle(SyncErrorDialog.kt:56)
	at com.ichi2.anki.dialogs.SyncErrorDialog.getNotificationTitle(SyncErrorDialog.kt:30)
	at com.ichi2.anki.AnkiActivity.showAsyncDialogFragment(AnkiActivity.kt:5)
	at com.ichi2.anki.DeckPicker.showSyncErrorDialog(DeckPicker.kt:3)
	at com.ichi2.anki.DeckPicker.showSyncErrorDialog(DeckPicker.kt:1)
	at com.ichi2.anki.SyncKt$createSyncListener$1.onPostExecute(Sync.kt:522)
	at com.ichi2.async.Connection.onPostExecute(Connection.kt:6)
	at com.ichi2.async.Connection.onPostExecute(Connection.kt:1)
	at android.os.AsyncTask.finish(AsyncTask.java:755)
	at android.os.AsyncTask.access$900(AsyncTask.java:192)
	at android.os.AsyncTask$InternalHandler.handleMessage(AsyncTask.java:772)
	at android.os.Handler.dispatchMessage(Handler.java:107)
	at android.os.Looper.loop(Looper.java:237)
	at android.app.ActivityThread.main(ActivityThread.java:8167)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:496)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1100)
```
Samsung 
Android 13
2.16beta01
https://ankidroid.org/acra/app/1/bug/47861/report/e4d607bb-64a5-42a9-ac71-e4a75d17b2b0

### PR DESCRIPTION
**Pull request by david-allison**
## Fixes
* Fixes #13795

## Approach
Determine that we have two problems:

1. We're using `AnkiDroidApp` resources as default from `res()`, this does not handle language changes well
2. We sometimes use `context.resources` inside `notificationMessage` or `notificationTitle`, this crashes when called in

https://github.com/ankidroid/Anki-Android/blob/08b6a0feeb634a70e99d3ec0887c003cbd81e238/AnkiDroid/src/main/java/com/ichi2/anki/AnkiActivity.kt#L413-L417

To fix (2), use `res()` throughout
To fix (1), prefer `resources` over `AnkiDroidApp.resources`

## How Has This Been Tested?
Unit tests

modify `sync` to always show the error dialog after a delay, and see the crashes

I haven't tested this thoroughly, I trusted the unit tests

## Checklist
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/DatabaseErrorDialog.kt [hunk 1]
```diff
@@ -493,7 +494,11 @@ class DatabaseErrorDialog : AsyncDialogFragment() {
                     databaseVersion
                 )
             }
-            DIALOG_STORAGE_UNAVAILABLE_AFTER_UNINSTALL -> res().getString(R.string.directory_inaccessible_after_uninstall_summary, CollectionHelper.getCurrentAnkiDroidDirectory(requireContext()))
+            DIALOG_STORAGE_UNAVAILABLE_AFTER_UNINSTALL -> {
+                val directory = context?.let { CollectionHelper.getCurrentAnkiDroidDirectory(it) }
+                    ?: res().getString(R.string.card_browser_unknown_deck_name)
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/dialogs/DatabaseErrorDialog.kt:499]: questionable, but such a rare case & it felt best to not crash

Comment (mikehardy) [reply AnkiDroid/src/main/java/com/ichi2/anki/dialogs/DatabaseErrorDialog.kt:499]: always best not to crash

**Review by mikehardy — APPROVED** [review]
I'm okay with all of it, this was a bedeviling crash, thanks for fixing it

**Review by lukstbit — APPROVED** [review]
Ok!

**Pull request merged**